src/sync.ts

```diff
@@ -70,46 +75,53 @@ export async function trySyncWithCredentials(gitHubUser: GitHubUser) {
  * Note: no concurrent calls!
  */
 export async function sync(myGitHubUser: GitHubUser) {
+  // Get existing data first
+  const existingRepoStateByFullName = await getRepoStateByFullName();
+
   const prBlocksAtSyncStart = await getMyPrBlockList();
   const reviewRequestBlocksAtSyncStart = await getReviewRequestBlockList();
   const commentBlocksAtSyncStart = await getCommentBlockList();
 
   resetGitHubCallsCounter();
   const syncStartUnixMillis = Date.now();
   const settings = await getSettings();
-  const allReposIncludingDisabled = await getRepos();
-  const repos = allReposIncludingDisabled.filter((v) => v.monitoringEnabled);
-  const prevRepoStateByFullName = await getRepoStateByFullName();
-  const repoStateByFullNameBuilder = new Map<string, RepoState>();
-
-  // used purely as a starting point (#NOT_MATURE: what if user unsubscribed from them?):
-  const recentNotifications = await listRecentNotifications(
-    settings.getMinCommentCreateDate(),
-  );
 
   const user = (await getUser()).data;
   myGitHubUser.login = user.login;
   const userTeams = await listUserTeams();
   myGitHubUser.teamIds = userTeams.map((v) => v.id);
 
+  const allReposIncludingDisabled = await getRepos();
+  const repos = allReposIncludingDisabled.filter((v) => v.monitoringEnabled);
+  const repoStateByFullNameBuilder = new Map<string, RepoState>(
+    existingRepoStateByFullName,
```

Comment:
I think that's wrong because the set of repos monitored might have changed in the meantime.